### PR TITLE
Add support for DICT popitem

### LINKED ISSUE #1887: [DICT] Add Python support for popitem
### Describe the feature
Syntactic and semantic support should be added for `popitem`. This should be tested by implementing Python printing

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
def foo():
    a = {1,1.0, 2:2.0}
    b = a.popitem()
    c = a.popitem()
    return a, b, c
```

### PR DESCRIPTION
**Pull request by shoaib-moeen**
Add Python support for the dict `popitem` method. Fixes #1887 

**Commit Summary**
- Add Python support for the dict `popitem` method.
- Fix the shape of an IndexedElement holding a 0D element of an inhomogeneous tuple
- Fix the creation of a variable from an assignment to a pyccel function returning an inhomogeneous tuple (related to #337)
- Raise an error instead of trying to return an inhomogeneous element (related to #337)
- Add tests

### CODE REVIEW

**Comment by shoaib-moeen** [comment]
/bot run docs spelling pylint pyccel_lint coverage

**Comment by shoaib-moeen** [comment]
/bot run docs spelling pylint pyccel_lint coverage

**Comment by EmilyBourne** [comment]
Never mind I have found how to reopen it myself

**Comment by EmilyBourne** [comment]
@shoaib-moeen Do you plan to come back and finish this PR or can I pick it up and complete it?

**Review by jalalium — COMMENTED** [review]
```
def pop_item():
    a = {1:1.0, 2:1.0}
    x = a.popitem()
    return x[0]
```
```
def pop_item():
    a = {1:1.0, 2:1.0}
    return a.popitem()[0]
```

Both of these snippets do not work.  Is this expected?

**Comment by EmilyBourne** [comment]
> ```
> def pop_item():
>     a = {1:1.0, 2:1.0}
>     x = a.popitem()
>     return x[0]
> ```
> 
> ```
> def pop_item():
>     a = {1:1.0, 2:1.0}
>     return a.popitem()[0]
> ```
> 
> Both of these snippets do not work. Is this expected?

XD no. I guess the result type must be wrong

**Comment by EmilyBourne** [comment]
Blocked by #1838

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -815,17 +815,29 @@ def is_int(a):
...
-                    return self._visit(var[indices[0]][indices[1:]])
+        elif hasattr(var,'__getitem__'):
+            if len(indices)==1:
+                return var[indices[0]]
             else:
-                var_type = type(var)
-                errors.report(f"Can't index {var_type}", symbol=expr,
-                    severity='fatal')
+                return self._visit(var[indices[0]][indices[1:]])
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/parser/semantic.py:827]: /bot accept
This problem existed before this branch

[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/parser/semantic.py:827]: See https://github.com/pyccel/pyccel/issues/1973

File: pyccel/parser/semantic.py [hunk 2]
```diff
@@ -815,17 +815,29 @@ def is_int(a):
...
+        elif isinstance(var, PyccelFunction):
+            pyccel_stage.set_stage('syntactic')
+            tmp_var = PyccelSymbol(self.scope.get_new_name())
+            assign = Assign(tmp_var, var)
+            assign.set_current_ast(expr.python_ast)
+            pyccel_stage.set_stage('semantic')
+            self._additional_exprs[-1].append(self._visit(assign))
+            var = self._visit(tmp_var)
+            print(var)
```
Comment (jalalium) [inline pyccel/parser/semantic.py:836]: I think this is a debugging print, right?

Comment (EmilyBourne) [reply pyccel/parser/semantic.py:836]: Good catch

Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: Fixed 25ede3f1

**Review by jalalium — APPROVED** [review]
Very good job!

**Pull request merged**